torch/csrc/distributed/c10d/Utils.cpp

```diff
@@ -72,15 +72,10 @@ std::vector<at::Tensor> getTensorShapes(
   std::vector<at::Tensor> shapeTensors;
   shapeTensors.reserve(tensors.size());
   for (const auto& tensor : tensors) {
-    auto shapesVec = tensor.sizes().vec();
-    int64_t shapes_size = shapesVec.size();
-    // Need to clone here otherwise the shapesVec.data() memory is not copied
-    // and can be released under the hood.
-    at::Tensor shapesTensor = at::from_blob(
-                                  shapesVec.data(),
-                                  {shapes_size},
-                                  at::TensorOptions().dtype(at::kLong))
-                                  .clone();
+    // Use `at::tensor()` to copy the data underlying `sizes()` since it may be
+    // released elsewhere.
+    at::Tensor shapesTensor =
```

Comment:
`at::tensor()` should make a copy: https://discuss.pytorch.org/t/can-i-initialize-tensor-from-std-vector-in-libtorch/33236/4

Reply:
Previously, `tensor.sizes().vec()` copies the data (underlying `tensor.sizes()`) once, and then `from_blob().clone()` copies the data a second time. This new approach _should_ save one copy.

Reply:
good catch!